tests/install_test.go

```diff
@@ -44,14 +44,23 @@ func eventuallyAssert(cmd string, m types.GomegaMatcher) {
 	}, 5*time.Minute, 10*time.Second).Should(m)
 }
 
+var vm VM
+
 var _ = Describe("kairos install test", Label("install-test"), func() {
 
 	BeforeEach(func() {
-		EventuallyConnects(1200)
+
+		vm = startVM()
+		vm.EventuallyConnects(1200)
 	})
 
 	AfterEach(func() {
 		Machine.Clean()
+		mc := Machine.Config()
+		port, _ := getFreePort()
+		mc.Display = fmt.Sprintf("-vga qxl -spice port=%d,addr=127.0.0.1,disable-ticketing=yes", port)
+		port, _ = getFreePort()
+		mc.SSH.Port = fmt.Sprintf("%d", port)
```

Comment:
wont this collide with the port above? After all its still free, as it hasnt been locked by anything yet?

Reply:
True, in seldom cases it will collide. Didn't think about this.

Reply:
wont this still potentially happens?

Let me see if I understand it correctly.

 - on line 60 you get a free port (one that its not currently used by anything).
 - then on line 62 you ask again for a free port.
 - then on line 64, the machine is created AND those ports are used at that moment (As soon as qemu/vbox locks those ports to be used on display/ssh)

What prevents `getFreePort` returning the same port on both calls? After all, for the kernel the port returned in the first calls is still free. So if you call it 100 times you may get it back 100 times the same port?
No matter that its opened by the `getFreePort` as its still closed as soon as the function returns.

So this can cause a collision by having both ports being the same one. I know, its rare but if it happens randomly that CI fails due to this we are gonna be puzzled.

I can show you the port collision does indeed happen: https://goplay.tools/snippet/qktKbufupVc
I got it to fail at the 2 port obtained, so its indeed a possibility that both ports returned are the same.

I would do a check for those and rerun `getFreePort` until I get a different port just in case :D

Reply:
You are still correct. It's still containing code from the previous strcuture. I have a nil pointer error which I still don't understand. So I haven't fixed that redundant code.